torch/csrc/jit/fuser/executor.cpp

```diff
@@ -262,17 +252,22 @@ void launchFusion(
   }
 
   // Adds (flattened) output arguments
+  outputs.reserve(fusion.outputDesc().size());
+  const auto ref_options = device >= 0 ?
+      inputs[0].options().device_index(device) :
+      inputs[0].options().device(at::Device{at::DeviceType::CPU});
```

Comment:
Shouldn’t the device of inputs already match the device of outputs?

Reply:
Probably; this was in the code before. Let me examine the code a little more to check this

Reply:
yes, you're right -- the kernel gets launched on the device of all inputs, so the device of the outputs should match the device of the inputs